core/src/main/java/io/questdb/cairo/sql/async/PageFrameReduceTask.java

```diff
@@ -0,0 +1,93 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2022 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.cairo.sql.async;
+
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.sql.StatefulAtom;
+import io.questdb.std.DirectLongList;
+import io.questdb.std.MemoryTag;
+import io.questdb.std.Misc;
+
+import java.io.Closeable;
+
+public class PageFrameReduceTask implements Closeable {
+    private final DirectLongList rows;
+    private final long pageFrameQueueCapacity;
+    private int frameIndex = Integer.MAX_VALUE;
+    private PageFrameSequence<?> frameSequence;
+
+    public PageFrameReduceTask(CairoConfiguration configuration, int pageFrameQueueCapacity) {
+        this.rows = new DirectLongList(configuration.getPageFrameReduceRowIdListCapacity(), MemoryTag.NATIVE_LONG_LIST);
```

Comment:
nit: would be nice to introduce a new tag for these lists. This would simplify diagnosing async offload memory footprint issues.

Reply:
done in https://github.com/questdb/questdb/pull/1732/commits/40648e75d7590eb17e3526b9b0cc362e1754a925